aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,303 @@
+
+> [!NOTE]
+> `skipRemainingRules: true` should be used whenever possible, because matching rules is an expensive process and slows down application response time. For the fastest application response, order your rewrite rules from most frequently matched to least frequently matched and skip the processing of the remaining rules when a match occurs and no additional rule processing is required.
+
+#### Apache mod_rewrite
+You can apply Apache mod_rewrite rules with `AddApacheModRewrite()`. The first parameter takes an `IFileProvider`, which is provided in the sample application via [Dependency Injection](dependency-injection.md) by injecting the `IHostingEnvironment` and using it to provide the `ContentRootFileProvider`. The second parameter is the path to your rules file, which is **ApacheModRewrite.txt** in the sample application. You must make sure that the rules file is deployed with the application. For more information and examples of mod_rewrite rules, see [Apache mod_rewrite](https://httpd.apache.org/docs/2.4/rewrite/).
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=4)]
+
+The sample application will redirect requests from **/apache-mod-rules-redirect/(.\*)** to **/redirected?id=$1**. The response status code is 302 (Found).
+
+[!code[Main](url-rewriting/sample/ApacheModRewrite.txt)]
+
+Original Request: **http://localhost:5000/apache-mod-rules-redirect/1234**
+
+![Browser window with Developer Tools tracking the requests and responses](url-rewriting/_static/add_apache_mod_redirect.png)
+
+##### Supported server variables
+The middleware supports the following Apache mod_rewrite server variables:
+* CONN_REMOTE_ADDR
+* HTTP_ACCEPT
+* HTTP_CONNECTION
+* HTTP_COOKIE
+* HTTP_FORWARDED
+* HTTP_HOST
+* HTTP_REFERER
+* HTTP_USER_AGENT
+* HTTPS
+* IPV6
+* QUERY_STRING
+* REMOTE_ADDR
+* REMOTE_PORT
+* REQUEST_FILENAME
+* REQUEST_METHOD
+* REQUEST_SCHEME
+* REQUEST_URI
+* SCRIPT_FILENAME
+* SERVER_ADDR
+* SERVER_PORT
+* SERVER_PROTOCOL
+* TIME
+* TIME_DAY
+* TIME_HOUR
+* TIME_MIN
+* TIME_MON
+* TIME_SEC
+* TIME_WDAY
+* TIME_YEAR
+
+#### IIS URL Rewrite Module rules
+To use rules that would normally apply to the IIS URL Rewrite Module, use `AddIISUrlRewrite()`. The first parameter takes an `IFileProvider`, while the second parameter is the path to your XML rules file, which is **IISUrlRewrite.xml** in the sample application. You must make sure that the rules file is deployed with the application. For more information and examples of IIS URL Rewrite Module rules, see [Using Url Rewrite Module 2.0](https://www.iis.net/learn/extensions/url-rewrite-module/using-url-rewrite-module-20) and [URL Rewrite Module Configuration Reference](https://www.iis.net/learn/extensions/url-rewrite-module/url-rewrite-module-configuration-reference).
```

Comment:
Do not point this at your web.config file, these rules should be stored outside of your web.config to avoid conflicts with the IIS Rewrite module.